heroku/heroku-vscode
# Jw/quick access menu

This PR adds the quick access menu to the app and the grammar for displaying themed heroku log content.

## To Test
1. Checkout this branch and run the extension. Make sure you are using node 20+
2. Open a heroku app sources project, click on the Heroku extension icon and authenticate (if necessary)
3. Hover on the tree item representing your app, you should see 3 menu items:
<img width="438" alt="image" src="https://github.com/user-attachments/assets/27303be2-ab18-4372-a591-0a7c448e19ef">

4. Right click on the same tree item an note the context menu:
<img width="222" alt="image" src="https://github.com/user-attachments/assets/f195db6b-fa4a-48fc-9ab0-40e1435fb01c">

5. Make sure each of these item are actionable. 

## Testing "Start a log session"
1. Choose the "Start a log session" item - you should see all Dynos being tailed.
2. Open the app in a browser and interact with it. You should see the logs update as requests are made

## Testing the grammar for themed Heroku log output
1. With an active log session displayed, note the color treatment applied to various facets of the log output:
<img width="1091" alt="image" src="https://github.com/user-attachments/assets/9a3e13e0-7e72-4d51-a497-9bd19516c0ac">

2. Note that this is similar to but not exactly like the colorized output when using `heroku logs -a <YOUR_APP> --tail --force-colors`
3. In vscode, go to settings -> settings: 
<img width="547" alt="image" src="https://github.com/user-attachments/assets/65e0a2e9-e985-4aa8-a176-25fe27665ed0">

4. choose workbench -> appearance and look for "color theme":
<img width="982" alt="image" src="https://github.com/user-attachments/assets/fff66542-b5fd-4903-a287-f6a6e261aae5">
5. Cycle through various themes and watch the Heroku log output adopt the chosen theme.

## Testing the grammar for themed Heroku log files
1. Copy all text from an active Heroku log output window and make a file named `heroku-output.log` 
2. paste the copied text to this file. 
3. Note that the log file adopts the same theme as the Heroku log output window
